servicetalk-data-jackson-jersey3-jakarta10/build.gradle

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2018-2019, 2021-2022, 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+apply plugin: "io.servicetalk.servicetalk-gradle-plugin-internal-library"
+
+checkstyleMain {
+  enabled false
+}
+
+checkstyleTest {
+  enabled false
+}
+
+spotbugsMain {
+  enabled false
+}
+
+task cleanSources(type: Delete) {
+  delete 'src'
+}
+
+task copySourcesForJersey3(type: Copy) {
+  dependsOn tasks.cleanSources
+  from '../servicetalk-data-jackson-jersey/src'
+  into 'src'
+  filter { line -> line.replaceAll('javax.ws.rs', 'jakarta.ws.rs') }
+  filter { line -> line.replaceAll('javax.inject', 'jakarta.inject') }
+
+  // Only modify specific javax.annotations.  Some like javax.annotation.Nullable need to stay the same
+  filter { line -> line.replaceAll('javax.annotation.Priority', 'jakarta.annotation.Priority') }
+}
+
+tasks.processResources.dependsOn(copySourcesForJersey3)
+tasks.sourcesJar.dependsOn(copySourcesForJersey3)
+tasks.compileJava.dependsOn(copySourcesForJersey3)
+tasks.processTestResources.dependsOn(copySourcesForJersey3)
+tasks.compileTestJava.dependsOn(copySourcesForJersey3)
+
+def actualJerseyVersion = "${jersey3VersionEE10}"
+def actualJavaxActivationVersion = "${javaxActivationVersionEE10}"
+def actualJavaxJaxbApiVersion = "${javaxJaxbApiVersionEE10}"
+def actualJavaxJaxbCoreVersion = "${javaxJaxbCoreVersionEE10}"
+def actualJavaxJaxbImplVersion = "${javaxJaxbImplVersionEE10}"
+def actualJaxRsVersion = "${jaxRsVersionEE10}"
+
+// Dependencies must remain consistent between servicetalk-data-jackson-jerseyX modules
+dependencies {
+  api platform(project(":servicetalk-dependencies"))
+  api platform("org.glassfish.jersey:jersey-bom:${actualJerseyVersion}")
```

Comment:
Note for other reviewers: we are at the limit of our gradle-fu. Please focus on correctness, documentation, and build behavior. Optimization of the build scripts and removal of the duplication might be a good task for the follow-up, but not a goal of this PR.